app/models/profile_rule.rb

```diff
@@ -2,10 +2,10 @@
 
 # Join table to be able to have a has-many-belongs-to-many relation between
 # Profile and Rule
-class ProfileRule < ApplicationRecord
+class ProfileRule < ProfileLink
   belongs_to :profile
-  belongs_to :rule
+  belongs_to :entity, :class_name => 'Rule', :foreign_key => :entity_id
 
   validates :profile, presence: true
-  validates :rule, presence: true, uniqueness: { scope: :profile }
+  validates :entity_id, presence: true, uniqueness: { scope: :profile }
```

Comment:
If you set up the `as:` above, this should not be changed at all.

Reply:
Okay, yeah good point. I updated the belongs_to to this:
`belongs_to :rule, class_name: 'Rule', foreign_key: :entity_id`
And the import is working correctly with the original `validates`